From 7462c0a9903ed71738404176b1d231c9c09fb7dd Mon Sep 17 00:00:00 2001
From: Jari Aalto <jari.aalto@cante.net>
Date: Wed, 30 Sep 2009 13:48:13 +0300
Subject: [PATCH] Makefile new toplevel

Signed-off-by: Jari Aalto <jari.aalto@cante.net>
---
 Makefile |   46 ++++++++++++++++++++++++++++++++++++++++++++++
 1 files changed, 46 insertions(+), 0 deletions(-)
 create mode 100755 Makefile

diff --git a/Makefile b/Makefile
new file mode 100755
index 0000000..4e6ac50
--- /dev/null
+++ b/Makefile
@@ -0,0 +1,46 @@
+#!/usr/bin/make -f
+#
+#	Copyright (C) 2009 Erwin Waterlander
+#	Copyright (C) 2009 Jari Aalto
+#
+#	This program is free software; you can redistribute it and/or
+#	modify it under the terms of the GNU General Public License as
+#	published by the Free Software Foundation; either version 2 of the
+#	License, or (at your option) any later version
+#
+#	This program is distributed in the hope that it will be useful, but
+#	WITHOUT ANY WARRANTY; without even the implied warranty of
+#	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+#	General Public License for more details.
+#
+#	Visit <http://www.gnu.org/copyleft/gpl.html>
+
+ifneq (,)
+This makefile requires GNU Make.
+endif
+
+PACKAGE = wcd
+OBJDIR	= src
+EXT	= .exe
+EXE	= $(OBJDIR)/$(PACKAGE)$(EXT)
+
+all: $(EXE)
+
+$(EXE):
+	$(MAKE) -C $(OBJDIR) CC="$(CC)"
+
+clean:
+	$(MAKE) -C $(OBJDIR) clean
+
+distclean:
+	$(MAKE) -C $(OBJDIR) distclean
+
+realclean:
+	$(MAKE) -C $(OBJDIR) realclean
+
+install: all
+	$(MAKE) -C $(OBJDIR) install
+
+.PHONY: clean distclean realclean install
+
+# End of file
-- 
1.6.1.2

