From d765ec2791997412779fa344ed622093afa57dbd Mon Sep 17 00:00:00 2001
From: Jari Aalto <jari.aalto@cante.net>
Date: Wed, 23 Jun 2010 19:45:38 +0300
Subject: [PATCH] src/Makefile: (%.pdf): Under Cygwin ignore status code 128
Organization: Private
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit


Signed-off-by: Jari Aalto <jari.aalto@cante.net>
---
 src/Makefile |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/src/Makefile
+++ b/src/Makefile
@@ -435,7 +435,7 @@ $(BIN): $(OBJS1)
 	$(CC) $(CFLAGS) $(EXTRA_CFLAGS) $(DEFS) -c $< -o $@
 
 %.pdf: %.ps
-	ps2pdf $< $@
+	ps2pdf $< $@ || :
 
 %.po : $(POT)
 	msgmerge -U $@ $(POT) --backup=numbered
